### ISSUE TITLE
posgres container does not start in Mac OSX

### ISSUE THREAD

**Issue by sertansenturk** [sertansenturk/cookiecutter-ds-docker#28]
@hedonistrh to fill the details

**One-line summary (optional)**

**Steps to Reproduce**
- Add screenshots to help explain your problem, if applicable
- Include specs (OS, Python version, code commit hash or release version), if useful

**Expected Behaviour**

**Design**

**Acceptance Criteria**

**Outcome**
- Clarify the deliverables (scripts, process etc.) and documentation, if applicable.

**Don't forget to**
- Point the ticket
- Set priority as a label
- Assign to a person
- Link dependencies, if applicable

**Comment by halil-erdogan** [comment]
**One Line Summary**
posgres error with the `make` command

**Steps to Reproduce**
- I run cookie-cutter command. 
- Then `make build` successfully built everything. 

> Successfully built 6785.....

- When I run `make`, I am getting error about postgres
Error is basically as following

> postgres_1  | initdb: could not look up effective user ID 501: user does not exist
> bocuk_postgres_1 exited with code 1
> postgres_1  | initdb: could not look up effective user ID 501: user does not exist
> bocuk_postgres_1 exited with code 1
> mlflow_1    | wait-for-it.sh: timeout occurred after waiting 15 seconds for postgres:5432
> postgres_1  | chmod: /var/run/postgresql: Operation not permitted
> postgres_1  | initdb: could not look up effective user ID 501: user does not exist
> bocuk_postgres_1 exited with code 1
> mlflow_1    | 2020/05/03 22:32:08 ERROR mlflow.cli: Error initializing backend store
> mlflow_1    | 2020/05/03 22:32:08 ERROR mlflow.cli: (psycopg2.OperationalError) could not translate host name "postgres" to address: Name or service not known
> mlflow_1    |
> mlflow_1    | (Background on this error at: http://sqlalche.me/e/e3q8)
> ....
> ....
> postgres_1  | initdb: could not look up effective user ID 501: user does not exist
> bocuk_postgres_1 exited with code 1
> postgres_1  | initdb: could not look up effective user ID 501: user does not exist
> bocuk_postgres_1 exited with code 1

When I investigate the Docker dashboard, this is what our GUI shows 
![image](https://user-images.githubusercontent.com/34948815/80927616-39f69800-8d9f-11ea-99c8-ddc292f1a24f.png)
**Info about My Setup**
I am using _OsX 10.15.4 (19E266)_ and my Docker Engine's version is _19.03.8_

**Comment by sertansenturk** [comment]
did some checks on my Mac; seems like a base image + racing condition problem.

will try to come up with a solution tomorrow...

**Comment by sertansenturk** [comment]
**TL:DR;** @hedonistrh can you run the command below and confirm if you can create and run a project successfully?

```bash
cookiecutter --checkout CC-28-postgres-does-not-start-in-mac-osx https://github.com/sertansenturk/cookiecutter-ds-docker
```

FYI, the problems and the solutions are described below:

For the first run, posgres needs to `initdb` with the correct user. We were following [the second solution (mounting `/etc/passwd` as read-only to the posgres container) suggested in the posgres docker documentation](https://github.com/docker-library/docs/tree/master/postgres#arbitrary---user-notes). Unfortunately, the user is not typically listed `/etc/passwd` (see the explanation [here](https://macosx.com/threads/users-and-etc-passwd.8848/)), so we were getting "a user does not exist error". Using [posgres debian image instead of alpine (first solution)](https://github.com/docker-library/docs/tree/master/postgres#arbitrary---user-notes), we can "fake" the `/etc/passwd` in the container. Commit ce0f3c8 fixed this problem.

Then, I stumbled upon a second problem: In mac osx, the posgres container takes significantly longer to bootstrap everything in the first run. `mlflow` was waiting for a maximum of 15 seconds, hence failing to connect to the database properly. Increasing the timeout to 1 minute fixes the problem (commit 3d7eb44).

As an extra precaution, ~I am going to [move the dbinit (third solution)](https://github.com/docker-library/docs/tree/master/postgres#arbitrary---user-notes) to `make ${POSTGRES_STORE}` (which is a prequisite of `make`), so that initialization can be decoupled from running the stack~, I made tests in the jupyter container wait until mlflow tracking server is ready (commit 8aaa466).
